quil-py/quil/program/__init__.pyi

```diff
@@ -238,6 +314,73 @@ class CalibrationSet:
         """
         ...
 
+
+@final
+class FixedScheduleItem:
+    """
+    A single item within a fixed schedule, representing a single instruction within a basic block.
+    """
+    @property
+    def instruction_index(self) -> int:
+        """
+        The index of the instruction within the basic block.
+        """
+    @property
+    def time_span(self) -> FixedTimeSpan:
+        """
+        The time span during which the instruction is scheduled.
+        """
+
+
+@final
+class ControlFlowGraph:
+    """
+    Representation of a control flow graph (CFG) for a Quil program.
+
+    The CFG is a directed graph where each node is a basic block and each edge is a control flow
+    transition between two basic blocks.
+    """
+
+    def has_dynamic_control_flow(self) -> bool: 
```

Comment:
From a DbC perspective, does `False` here imply `len(self.basic_blocks()) == 1`? If yes, that would be worth documenting.

Reply:
I added a clarifying note

Reply:
Note is good!